engines/bastion_katello/app/assets/javascripts/bastion_katello/products/products.routes.js

```diff
@@ -42,13 +42,14 @@ angular.module('Bastion.products').config(['$stateProvider', function ($statePro
     });
 
     $stateProvider.state("product-discovery", {
-        abstract: true,
         url: '/products/discovery',
-        controller: 'DiscoveryController',
-        templateUrl: 'products/discovery/views/discovery-base.html'
+        templateUrl: 'products/discovery/views/discovery-base.html',
+        permission: 'edit_products',
+        redirectTo: 'product-discovery.scan'
     })
     .state("product-discovery.scan", {
         url: '/scan',
+        controller: 'DiscoveryController',
```

Comment:
This seemed to do the trick w.r.t the browser console error. I can still discover and enable repos by hitting the page directly or navigating from /products list